File: generated_tests/csharp/tests/camera_tests.cs
using Xunit;
using static SplashKitSDK.SplashKit;

namespace SplashKitTests
{
    public class IntegrationTests
    {
        [Fact]
        public void TestCameraPositionIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            var testCameraPosition = CameraPosition();
            Assert.Equal(testCameraPosition, 0.0);
            Assert.Equal(testCameraPosition, 0.0);
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestCameraXIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            SetCameraPosition(PointAt(100.0, 100.0));
            Assert.Equal(100.0, CameraX());
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestCameraYIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            SetCameraPosition(PointAt(100.0, 200.0));
            Assert.Equal(200.0, CameraY());
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestCenterCameraOnVectorIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            var testSprite = CreateSprite("test_sprite");
            SpriteSetPosition(testSprite, PointAt(100.0, 100.0));
            CenterCameraOn(testSprite, VectorFromAngle(50.0, 50.0));
            Assert.Equal(PointAt(50.0, 50.0), CameraPosition());
            FreeSprite(testSprite);
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestCenterCameraOnIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            var testSprite = CreateSprite("test_sprite");
            SpriteSetPosition(testSprite, PointAt(100.0, 100.0));
            CenterCameraOn(testSprite, 0.0, 0.0);
            Assert.Equal(PointAt(50.0, 50.0), CameraPosition());
            FreeSprite(testSprite);
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestMoveCameraByVectorIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            var testVector = VectorFromAngle(0.0, 100.0);
            MoveCameraBy(testVector);
            Assert.Equal(100.0, CameraX());
            Assert.Equal(100.0, CameraY());
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestMoveCameraByIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            MoveCameraBy(100.0, 100.0);
            Assert.Equal(100.0, CameraX());
            Assert.Equal(100.0, CameraY());
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestMoveCameraToPointIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            MoveCameraTo(PointAt(100.0, 100.0));
            Assert.Equal(PointAt(100.0, 100.0), CameraPosition());
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestMoveCameraToIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            MoveCameraTo(100.0, 100.0);
            Assert.Equal(100.0, CameraX());
            Assert.Equal(100.0, CameraY());
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestPointInWindowIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            var testPoint = PointAt(400, 300);
            Assert.True(PointInWindow(testWindow, testPoint));
            var testPointOutside = PointAt(1000, 1000);
            Assert.False(PointInWindow(testWindow, testPointOutside));
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestPointOnScreenIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            var testPoint = PointAt(400.0, 300.0);
            Assert.True(PointOnScreen(testPoint));
            var testPointOutside = PointAt(1000.0, 1000.0);
            Assert.False(PointOnScreen(testPointOutside));
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestRectInWindowIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            var testRectangle = RectangleFrom(0, 0, 100, 100);
            Assert.True(RectInWindow(testWindow, testRectangle));
            var testRectangleOutside = RectangleFrom(1000, 1000, 100, 100);
            Assert.False(RectInWindow(testWindow, testRectangleOutside));
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestRectOnScreenIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            var testRectangle = RectangleFrom(0, 0, 100, 100);
            Assert.True(RectOnScreen(testRectangle));
            MoveCameraTo(1000, 1000);
            Assert.False(RectOnScreen(testRectangle));
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestScreenCenterIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            var testCenter = ScreenCenter();
            Assert.Equal(testCenter, Todo());
            Assert.Equal(testCenter, Todo());
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestScreenRectangleIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            var testRectangle = ScreenRectangle();
            Assert.Equal(800, testRectangle.Width);
            Assert.Equal(600, testRectangle.Height);
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestSetCameraPositionIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            SetCameraPosition(PointAt(100.0, 100.0));
            Assert.Equal(PointAt(100.0, 100.0), CameraPosition());
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestSetCameraXIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            SetCameraX(100.0);
            Assert.Equal(100.0, CameraX());
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestSetCameraYIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            SetCameraY(100.0);
            Assert.Equal(100.0, CameraY());
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestToScreenPointIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            SetCameraPosition(PointAt(100.0, 100.0));
            var testScreenPoint = ToScreen(PointAt(150.0, 150.0));
            Assert.Equal(testScreenPoint, Todo());
            Assert.Equal(testScreenPoint, Todo());
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestToScreenRectangleIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            var testRectangle = RectangleFrom(100.0, 100.0, 200.0, 200.0);
            var screenRectangle = ToScreen(testRectangle);
            Assert.Equal(ToScreenX(100.0), RectangleLeft(screenRectangle));
            Assert.Equal(ToScreenY(100.0), RectangleTop(screenRectangle));
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestToScreenXIntegration()
        {
            SetCameraX(100.0);
            var testScreenX = ToScreenX(150.0);
            Assert.Equal(50.0, testScreenX);
        }
        [Fact]
        public void TestToScreenYIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            SetCameraPosition(PointAt(100.0, 100.0));
            var testScreenY = ToScreenY(150.0);
            Assert.Equal(50.0, testScreenY);
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestToWorldIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            SetCameraPosition(PointAt(100.0, 100.0));
            var testWorldPoint = ToWorld(PointAt(400.0, 300.0));
            Assert.Equal(testWorldPoint, Todo());
            Assert.Equal(testWorldPoint, Todo());
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestToWorldXIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            SetCameraPosition(PointAt(100.0, 100.0));
            var testWorldX = ToWorldX(400.0);
            Assert.Equal(CameraX(), testWorldX);
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestToWorldYIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            SetCameraPosition(PointAt(100.0, 100.0));
            var testWorldY = ToWorldY(300.0);
            Assert.Equal(400.0, testWorldY);
            CloseWindow(testWindow);
        }
        [Fact]
        public void TestVectorWorldToScreenIntegration()
        {
            var testVector = VectorWorldToScreen();
            Assert.NotEqual(VectorFromAngle(0.0, 0.0), testVector);
        }
        [Fact]
        public void TestWindowAreaIntegration()
        {
            var testWindow = OpenWindow("Test Window", 800, 600);
            var testArea = WindowArea(testWindow);
            Assert.Equal(800, testArea.Width);
            Assert.Equal(600, testArea.Height);
            CloseWindow(testWindow);
        }
    }
}

File: generated_tests/rust/tests/camera_tests.rs
use splashkit::*;
#[cfg(test)]
mod test_runner {
    pub fn run_tests_sequential(tests: &[&dyn Fn()]) {
        for test in tests {
            test();
        }
    }
}
#![test_runner(test_runner::run_tests_sequential)]
#[test]
fn test_camera_position_integration() {
    let test_window = open_window("Test Window", 800, 600);
    let test_camera_position = camera_position();
    assert_eq!(0.0, test_camera_position);
    assert_eq!(0.0, test_camera_position);
    close_window(test_window);
}
#[test]
fn test_camera_x_integration() {
    let test_window = open_window("Test Window", 800, 600);
    set_camera_position(point_at(100.0, 100.0));
    assert_eq!(camera_x(), 100.0);
    close_window(test_window);
}
#[test]
fn test_camera_y_integration() {
    let test_window = open_window("Test Window", 800, 600);
    set_camera_position(point_at(100.0, 200.0));
    assert_eq!(camera_y(), 200.0);
    close_window(test_window);
}
#[test]
fn test_center_camera_on_vector_integration() {
    let test_window = open_window("Test Window", 800, 600);
    let test_sprite = create_sprite("test_sprite");
    sprite_set_position(test_sprite, point_at(100.0, 100.0));
    center_camera_on_vector(test_sprite, vector_from_angle(50.0, 50.0));
    assert_eq!(camera_position(), point_at(50.0, 50.0));
    free_sprite(test_sprite);
    close_window(test_window);
}
#[test]
fn test_center_camera_on_integration() {
    let test_window = open_window("Test Window", 800, 600);
    let test_sprite = create_sprite("test_sprite");
    sprite_set_position(test_sprite, point_at(100.0, 100.0));
    center_camera_on(test_sprite, 0.0, 0.0);
    assert_eq!(camera_position(), point_at(50.0, 50.0));
    free_sprite(test_sprite);
    close_window(test_window);
}
#[test]
fn test_move_camera_by_vector_integration() {
    let test_window = open_window("Test Window", 800, 600);
    let test_vector = vector_from_angle(0.0, 100.0);
    move_camera_by_vector(test_vector);
    assert_eq!(camera_x(), 100.0);
    assert_eq!(camera_y(), 100.0);
    close_window(test_window);
}
#[test]
fn test_move_camera_by_integration() {
    let test_window = open_window("Test Window", 800, 600);
    move_camera_by(100.0, 100.0);
    assert_eq!(camera_x(), 100.0);
    assert_eq!(camera_y(), 100.0);
    close_window(test_window);
}
#[test]
fn test_move_camera_to_point_integration() {
    let test_window = open_window("Test Window", 800, 600);
    move_camera_to_point(point_at(100.0, 100.0));
    assert_eq!(camera_position(), point_at(100.0, 100.0));
    close_window(test_window);
}
#[test]
fn test_move_camera_to_integration() {
    let test_window = open_window("Test Window", 800, 600);
    move_camera_to(100.0, 100.0);
    assert_eq!(camera_x(), 100.0);
    assert_eq!(camera_y(), 100.0);
    close_window(test_window);
}
#[test]
fn test_point_in_window_integration() {
    let test_window = open_window("Test Window", 800, 600);
    let test_point = point_at(400, 300);
    assert!(point_in_window(test_window, test_point));
    let test_point_outside = point_at(1000, 1000);
    assert!(!point_in_window(test_window, test_point_outside));
    close_window(test_window);
}
#[test]
fn test_point_on_screen_integration() {
    let test_window = open_window("Test Window", 800, 600);
    let test_point = point_at(400.0, 300.0);
    assert!(point_on_screen(test_point));
    let test_point_outside = point_at(1000.0, 1000.0);
    assert!(!point_on_screen(test_point_outside));
    close_window(test_window);
}
#[test]
fn test_rect_in_window_integration() {
    let test_window = open_window("Test Window", 800, 600);
    let test_rectangle = rectangle_from(0, 0, 100, 100);
    assert!(rect_in_window(test_window, test_rectangle));
    let test_rectangle_outside = rectangle_from(1000, 1000, 100, 100);
    assert!(!rect_in_window(test_window, test_rectangle_outside));
    close_window(test_window);
}
#[test]
fn test_rect_on_screen_integration() {
    let test_window = open_window("Test Window", 800, 600);
    let test_rectangle = rectangle_from(0, 0, 100, 100);
    assert!(rect_on_screen(test_rectangle));
    move_camera_to(1000, 1000);
    assert!(!rect_on_screen(test_rectangle));
    close_window(test_window);
}
#[test]
fn test_screen_center_integration() {
    let test_window = open_window("Test Window", 800, 600);
    let test_center = screen_center();
    assert_eq!(todo(), test_center);
    assert_eq!(todo(), test_center);
    close_window(test_window);
}
#[test]
fn test_screen_rectangle_integration() {
    let test_window = open_window("Test Window", 800, 600);
    let test_rectangle = screen_rectangle();
    assert_eq!(test_rectangle.width, 800);
    assert_eq!(test_rectangle.height, 600);
    close_window(test_window);
}
#[test]
fn test_set_camera_position_integration() {
    let test_window = open_window("Test Window", 800, 600);
    set_camera_position(point_at(100.0, 100.0));
    assert_eq!(camera_position(), point_at(100.0, 100.0));
    close_window(test_window);
}
#[test]
fn test_set_camera_x_integration() {
    let test_window = open_window("Test Window", 800, 600);
    set_camera_x(100.0);
    assert_eq!(camera_x(), 100.0);
    close_window(test_window);
}
#[test]
fn test_set_camera_y_integration() {
    let test_window = open_window("Test Window", 800, 600);
    set_camera_y(100.0);
    assert_eq!(camera_y(), 100.0);
    close_window(test_window);
}
#[test]
fn test_to_screen_point_integration() {
    let test_window = open_window("Test Window", 800, 600);
    set_camera_position(point_at(100.0, 100.0));
    let test_screen_point = to_screen_point(point_at(150.0, 150.0));
    assert_eq!(todo(), test_screen_point);
    assert_eq!(todo(), test_screen_point);
    close_window(test_window);
}
#[test]
fn test_to_screen_rectangle_integration() {
    let test_window = open_window("Test Window", 800, 600);
    let test_rectangle = rectangle_from(100.0, 100.0, 200.0, 200.0);
    let screen_rectangle = to_screen_rectangle(test_rectangle);
    assert_eq!(rectangle_left(screen_rectangle), to_screen_x(100.0));
    assert_eq!(rectangle_top(screen_rectangle), to_screen_y(100.0));
    close_window(test_window);
}
#[test]
fn test_to_screen_x_integration() {
    set_camera_x(100.0);
    let test_screen_x = to_screen_x(150.0);
    assert_eq!(test_screen_x, 50.0);
}
#[test]
fn test_to_screen_y_integration() {
    let test_window = open_window("Test Window", 800, 600);
    set_camera_position(point_at(100.0, 100.0));
    let test_screen_y = to_screen_y(150.0);
    assert_eq!(test_screen_y, 50.0);
    close_window(test_window);
}
#[test]
fn test_to_world_integration() {
    let test_window = open_window("Test Window", 800, 600);
    set_camera_position(point_at(100.0, 100.0));
    let test_world_point = to_world(point_at(400.0, 300.0));
    assert_eq!(todo(), test_world_point);
    assert_eq!(todo(), test_world_point);
    close_window(test_window);
}
#[test]
fn test_to_world_x_integration() {
    let test_window = open_window("Test Window", 800, 600);
    set_camera_position(point_at(100.0, 100.0));
    let test_world_x = to_world_x(400.0);
    assert_eq!(test_world_x, camera_x());
    close_window(test_window);
}
#[test]
fn test_to_world_y_integration() {
    let test_window = open_window("Test Window", 800, 600);
    set_camera_position(point_at(100.0, 100.0));
    let test_world_y = to_world_y(300.0);
    assert_eq!(test_world_y, 400.0);
    close_window(test_window);
}
#[test]
fn test_vector_world_to_screen_integration() {
    let test_vector = vector_world_to_screen();
    assert_ne!(test_vector, vector_from_angle(0.0, 0.0));
}
#[test]
fn test_window_area_integration() {
    let test_window = open_window("Test Window", 800, 600);
    let test_area = window_area(test_window);
    assert_eq!(test_area.width, 800);
    assert_eq!(test_area.height, 600);
    close_window(test_window);
}